js/compiler/compiler.go

```diff
@@ -194,16 +243,61 @@ func newBabel() (*babel, error) {
 	return result, err
 }
 
-// Transform the given code into ES5, while synchronizing to ensure only a single
+// increaseMappingsByOne increases the lines in the sourcemap by line so that it fixes the case where we need to wrap a
+// required file in a function to support/emulate commonjs
+func (c *compilationState) increaseMappingsByOne(sourceMap []byte) ([]byte, error) {
+	var err error
+	m := make(map[string]interface{})
+	if err = json.Unmarshal(sourceMap, &m); err != nil {
+		return nil, err
+	}
+	mappings, ok := m["mappings"]
+	if !ok {
+		// no mappings, no idea what this will do, but just return it as technically we can have sourcemap with sections
+		// TODO implement incrementing of `offset` in the sections?
+		// TODO (kind of alternatively) drop the newline in the "commonjs" wrapping and have only the first line wrong
```

Comment:
> // no mappings, no idea what this will do, but just return it as technically we can have sourcemap with sections  
 		
and 

> // TODO implement incrementing of `offset` in the sections? 

are connected, sections is part of an *extension* of the sourcemap [specification](https://sourcemaps.info/spec.html#h.n05z8dfyl3yh). 

> // TODO (kind of alternatively) drop the newline in the "commonjs" wrapping and have only the first line wrong

Has to do with just removing the whole function here. 

> but it's not clear what kind of problems the current approach here can result in ?

Given that we don't do anything to the sourcemap we don't fix the off by one error commonjs wrapping adds - see the newly added documentation of the method.

Reply:
Added some more explanations in 51b7df6

Reply:
Okay, I think I got it. It's not about "fixing smth" but more about potential enhancements. Thanks for the update :+1: